service/src/test/java/uk/gov/hmcts/reform/fpl/service/RespondentServiceTest.java

```diff
@@ -0,0 +1,259 @@
+
+        AboutToStartOrSubmitCallbackResponse response = service.addHiddenValues(caseDetails);
+
+        assertThat(response.getData()).isEqualTo(caseDetails.getData());
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void shouldKeepExistingPartyIDWhenAlreadyExists() {
+        Map<String, Object> respondentObject = new HashMap<>();
+
+        respondentObject.put("respondents1", ImmutableList.of(
+            ImmutableMap.of(
+                "id", "12345",
+                "value", ImmutableMap.of(
+                    "party", RespondentParty.builder()
+                        .partyID("123")
+                        .build()
+                ))));
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(respondentObject)
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = service.addHiddenValues(caseDetails);
+
+        Map<String, Object> data = response.getData();
+        List<Map<String, Object>> respondents = (List<Map<String, Object>>) data.get("respondents1");
+        Map<String, Object> value = (Map<String, Object>) respondents.get(0).get("value");
+        Map<String, Object> party = (Map<String, Object>) value.get("party");
+
+        assertThat(party.get("partyID")).isEqualTo("123");
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void shouldKeepExistingPartyIdAndContinueAddingNewPartyId() {
+        Map<String, Object> respondentObject = new HashMap<>();
+
+        respondentObject.put("respondents1", ImmutableList.of(
+            ImmutableMap.of(
+                "id", "12345",
+                "value", ImmutableMap.of(
+                    "party", RespondentParty.builder()
+                        .firstName("James")
+                        .partyID("123")
+                        .build()
+                )),
+            ImmutableMap.of(
+                "id", "98765",
+                "value", ImmutableMap.of(
+                    "party", RespondentParty.builder()
+                        .firstName("Lucy")
+                        .build()
+                ))));
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(respondentObject)
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = service.addHiddenValues(caseDetails);
+
+        Map<String, Object> data = response.getData();
+        List<Map<String, Object>> respondents = (List<Map<String, Object>>) data.get("respondents1");
+        Map<String, Object> firstValue = (Map<String, Object>) respondents.get(0).get("value");
+        Map<String, Object> secondValue = (Map<String, Object>) respondents.get(1).get("value");
+        Map<String, Object> firstParty = (Map<String, Object>) firstValue.get("party");
+        Map<String, Object> secondParty = (Map<String, Object>) secondValue.get("party");
+
+        assertThat(firstParty).containsEntry("firstName", "James");
+        assertThat(firstParty.get("partyID")).isNotNull();
```

Comment:
If it is important to assure that new ID is added while old one is preserved and old one is known there should be an assertion to make sure existing ID is not overridden.

Reply:
I have this unit test to verify this   

void shouldKeepExistingPartyIDWhenAlreadyExists() {

Reply:
I understand that test I question says `should keep existing party id and continue adding new party id`. That suggests that keeping existing ID for first entry is important and due to that there should be assertion to guarantee it.

Reply:
I see what you're saying now, yes the first assertion should be asserting the id I have set. Made the change 👍